### DISCUSSION TITLE
IconFinder made my view cache directory a lot of cached blade files. Filesystem is on fire. Help.

### DISCUSSION THREAD

**Discussion by macit-emre** [orchidsoftware/platform#2820] [Q&A]
I used icons pretty much in my project. In live env, the blade view cache directory is struggling to find the file as there are 50k+ files there. 
I need to optimize that directory. Any idea will be appreciated.

Is there a way to use icons, as a normal css icon? (Without a custom implementation?) or if there  may be a custom implementation, how is this possible, without changing my entire project?

**Comment by tabuna** [comment]
Could you please try version 4.1.1 and then clear the templates to see if the mentioned issue persists?

**Reply by macit-emre** [reply]
i cant just update composer on live, but, i will try in my preprod env, then schedule an update for live.

By the way, Cant believe that you just updated the source code with my ticket. <3

**Reply by tabuna** [reply]
After checking on a small project at https://laravel.su, I noticed a decrease in the number of templates from 823 to 231. Furthermore, I no longer detect files starting with `<svg>`. I think the same will apply to you.

**Comment by macit-emre** [comment]
i did upgrade to 4.1.1 but this time, filesystem performance dropped because of pulling svg file in every request and not caching it to as it was auto caching in php-fpm. This is a "static file dependancy in software" problem.
i use an nfs system and all i want is changing cache directory of icon component cache.

clearing the crowd in view cache directory is still good but, i have this situation now :)

**Reply by tabuna** [reply]
In that case, I suggest considering the possibility of not inserting inline elements and instead creating an SVG sprite, for example:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg">
    <defs>
        <symbol viewBox="0 0 20 20" id="name-icon">
            <path d="..."/>
        </symbol>
    </defs>
</svg>
```

And when using it:

```xml
<svg viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
  <a xlink:href="https://example.com/icons?name=boostrap#name-icon"></a>
</svg>
```

But then we would need to add a public route that would assemble the sprite. What do you think about this?

**Reply by macit-emre** [reply]
i appreciate that, but this will increase non-cached request count to 20-30 x . 

My project is an enterprise project that already have 8k users in 1 year. We have ratelimiting mechanism, firewalls, etc. We also use cloudflare. In that case, ratelimiting mechanism will challenge us. And if we define a rule exceptional for this route, we will be also much more sensitive to DDOS attacks to that route. 
What i need is using css icons with this library if possible. Or as i mention, optimising view directory with previous version of blade-icons. 

You dont really have to find me a solution, i appreciate your exertion. Also big projects with big traffic (like mine's) offers a way to strengthen your framework when you offer robust solutions. Thanks a lot.

**Reply by macit-emre** [reply]
-> optimising view directory with previous version of blade-icons.
by the way, i think this is a laravel case , not an orchid case.
